packages/fields/src/components/media-edit/index.tsx

```diff
@@ -0,0 +1,299 @@
+			role="button"
+			tabIndex={ 0 }
+			onClick={ open }
+			onKeyDown={ ( event ) => {
+				if ( event.key === 'Enter' || event.key === ' ' ) {
+					event.preventDefault();
+					open();
+				}
+			} }
+			aria-label={ label }
+		>
+			{ children }
+		</div>
+	);
+}
+
+const archiveMimeTypes = [
+	'application/zip',
+	'application/x-zip-compressed',
+	'application/x-rar-compressed',
+	'application/x-7z-compressed',
+	'application/x-tar',
+	'application/x-gzip',
+];
+
+function MediaPreview( {
+	url,
+	attachment,
+}: {
+	url: string;
+	attachment: Attachment< 'view' > | null;
+} ) {
+	if ( ! attachment ) {
+		return null;
+	}
+	const attachmentTitle = attachment.title.rendered;
+	const mimeType = attachment.mime_type;
+	let preview: JSX.Element = <Icon icon={ file } />;
+	if ( mimeType.startsWith( 'image/' ) ) {
+		preview = (
+			<img
+				className="fields__media-edit-thumbnail"
+				alt={ attachment.alt_text || '' }
+				src={ url }
+			/>
+		);
+	} else if ( mimeType.startsWith( 'audio/' ) ) {
+		preview = <Icon icon={ audio } />;
+	} else if ( mimeType.startsWith( 'video/' ) ) {
+		preview = <Icon icon={ video } />;
+	} else if ( archiveMimeTypes.includes( mimeType ) ) {
+		preview = <Icon icon={ archive } />;
+	}
+	return (
+		<>
+			{ preview }
+			<Truncate
+				className="fields__media-edit-filename"
+				title={ attachmentTitle }
+			>
+				{ attachmentTitle }
+			</Truncate>
+		</>
+	);
+}
+
+/**
+ * A media edit control component that provides a media picker UI with upload functionality
+ * for selecting WordPress media attachments. Supports both the traditional WordPress media
+ * library and the experimental DataViews media modal.
+ *
+ * This component is intended to be used as the `Edit` property of a field definition when
+ * registering fields with `registerEntityField` from `@wordpress/editor`.
+ *
+ * @template Item - The type of the item being edited.
+ *
+ * @param {MediaEditProps<Item>} props                - The component props.
+ * @param {Item}                 props.data           - The item being edited.
+ * @param {Object}               props.field          - The field configuration with getValue and setValue methods.
+ * @param {Function}             props.onChange       - Callback function when the media selection changes.
+ * @param {string[]}             [props.allowedTypes] - Array of allowed media types. Default `['image']`.
+ * @param {boolean}              [props.multiple]     - Whether to allow multiple media selections. Default `false`.
+ *
+ * @return {JSX.Element} The media edit control component.
+ *
+ * @example
+ * ```tsx
+ * import { MediaEdit } from '@wordpress/fields';
+ * import type { MediaEditProps } from '@wordpress/fields';
+ *
+ * const featuredImageField = {
+ *   id: 'featured_media',
+ *   type: 'media',
+ *   label: 'Featured Image',
+ *   Edit: (props: MediaEditProps<MyPostType>) => (
+ *     <MediaEdit
+ *       {...props}
+ *       allowedTypes={['image']}
+ *     />
+ *   ),
```

Comment:
The edit component would receive the `DataFormControlProps`, not `MediaEditProps`:

```suggestion
 *   Edit: (props: DataFormControlProps<MyPostType>) => (
 *     <MediaEdit
 *       {...props}
 *       allowedTypes={['image']}
 *     />
 *   ),
```